toolkit/tools/build_test_cases/4/Readme.txt

```diff
@@ -0,0 +1,34 @@
+Dependency on a dynamically-generated provides. Similar to tests suite 2.
```

Comment:
Permutation of test 4+6:
Local copy of A dynamic BRs C
Upstream copy of A is unchanged, but does not list BR C (maybe macro environment changed?)
We have changed C locally

If we have nothing locally, I'd expect us to re-use A and avoid rebuilding
If we have a local copy of the new A, I'd expect us to now rebuild C -> A